Model/CardRepository.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Paytrail\PaymentService\Model;
+
+use Magento\Authorization\Model\UserContextInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Vault\Api\PaymentTokenRepositoryInterface;
+use Paytrail\PaymentService\Api\CardRepositoryInterface;
+use Paytrail\PaymentService\Helper\ApiData;
+
+class CardRepository implements CardRepositoryInterface
+{
+    /**
+     * @var ApiData
+     */
+    private ApiData $apiData;
+
+    /**
+     * @var PaymentTokenRepositoryInterface
+     */
+    private PaymentTokenRepositoryInterface $paymentTokenRepository;
+
+    /**
+     * @var UserContextInterface
+     */
+    private UserContextInterface $userContext;
+
+    /**
+     * @param ApiData $apiData
+     * @param PaymentTokenRepositoryInterface $paymentTokenRepository
+     * @param UserContextInterface $userContext
+     */
+    public function __construct(
+        ApiData $apiData,
+        PaymentTokenRepositoryInterface $paymentTokenRepository,
+        UserContextInterface $userContext
+    ) {
+        $this->apiData = $apiData;
+        $this->paymentTokenRepository = $paymentTokenRepository;
+        $this->userContext = $userContext;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function save(): string
```

Comment:
Misleading naming convention:
You're not saving anything during this request. You're getting the url address of the add to card form. Consider renaming this function